common/persistence/nosql/nosql_execution_store_util_test.go

```diff
@@ -0,0 +1,61 @@
+package nosql
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/uber/cadence/common"
+	"testing"
+
+	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/persistence/nosql/nosqlplugin"
+)
+
+func TestNosqlExecutionStoreUtils(t *testing.T) {
+	testCases := []struct {
+		name       string
+		setupStore func(*nosqlExecutionStore) (*nosqlplugin.WorkflowExecutionRequest, error)
+		input      *persistence.InternalWorkflowSnapshot
+		validate   func(*testing.T, *nosqlplugin.WorkflowExecutionRequest, error)
+	}{
+		{
+			name: "PrepareCreateWorkflowExecutionRequestWithMaps - Success",
+			setupStore: func(store *nosqlExecutionStore) (*nosqlplugin.WorkflowExecutionRequest, error) {
+				workflowSnapshot := &persistence.InternalWorkflowSnapshot{
+					ExecutionInfo: &persistence.InternalWorkflowExecutionInfo{
+						DomainID:   "test-domain-id",
+						WorkflowID: "test-workflow-id",
+						RunID:      "test-run-id",
+					},
+					VersionHistories: &persistence.DataBlob{
+						Encoding: common.EncodingTypeJSON,
+						Data:     []byte(`[{"Branches":[{"BranchID":"test-branch-id","BeginNodeID":1,"EndNodeID":2}]}]`),
+					},
+				}
+				return store.prepareCreateWorkflowExecutionRequestWithMaps(workflowSnapshot)
+			},
+			input: &persistence.InternalWorkflowSnapshot{
+				// Populate this as needed for the test
+			},
+			validate: func(t *testing.T, req *nosqlplugin.WorkflowExecutionRequest, err error) {
```

Comment:
if `validate` callback implementation will look similar for different test cases consider moving it to test body below

Reply:
Will make the changes while including more test cases.